### PR TITLE
disable regular STUNs on inactivity

### PR DESCRIPTION
**Pull request by bradfitz**
This change is [Reviewable](https://reviewable.io/reviews/tailscale/tailscale/511)

### CODE REVIEW

**Comment by bradfitz** [comment]
Okay, this works for me now. Ready for review.

**Comment by bradfitz** [comment]
(This also lays the groundwork for pushing NetworkMap down into magicsock so I can grab out the DiscoKeys)

File: wgengine/magicsock/magicsock.go [hunk 1]
```diff
@@ -1487,12 +1544,22 @@ func (c *Conn) periodicReSTUN() {
 	}
 	timer := time.NewTimer(dur())
 	defer timer.Stop()
+	var lastIdleState interface{} // nil or bool
```
Comment (crawshaw) [inline wgengine/magicsock/magicsock.go:1547]: *bool?

Comment (bradfitz) [reply wgengine/magicsock/magicsock.go:1547]: actually I remembered we had the opt.Bool type, so used that instead.

**Pull request merged**